lib/pure/times.nim

```diff
@@ -526,9 +554,58 @@ when not defined(JS):
     var cTimeInfo = timeInfo # for C++ we have to make a copy,
     # because the header of mktime is broken in my version of libc
     result = mktime(timeInfoToTM(cTimeInfo))
+    when defined(windows):
+      var secondOffset: TimeImpl = 0
+      if result == Time(-1) and cTimeInfo.year <= 1970:
```

Comment:
Assumptions I make:

 * `mktime` on Windows returns `-1` if time interpreted in local timezone predates 1970. I did not find a document that directly describes that, it has been determined by testing.
 * Gregorian calendar has been in use since 15.10.1582; previous day was 4.10.1582 in Julian calendar. Julian calender has one leap year on every year divisible by 4, Gregorian calendar additionally dumps those divisible by 100 unless they are also divisible by 400. These assumptions all come from Wikipedia.
 * Another assumption is that the given date is actually valid, i.e. a valid Gregorian or Julian date depending on whether it is before or after 15.10.1582.